src/main/java/com/redhat/devtools/lsp4ij/LSPFileListener.java

```diff
@@ -153,6 +153,33 @@ private FileEvent fe(URI uri, FileChangeType type) {
         return uri;
     }
 
+    /**
+     * Implements the <a href=https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_didRename">
+     * LSP specification for document renaming</a>. Used for both file renames within the same directory and file shifts
+     * to a different directory.
+     *
+     * @param virtualParentFile directory of the file after renaming
+     * @param oldFileName file name before renaming
+     * @param virtualNewFile virtual file after renaming
+     * @return URI of the file before renaming
+     */
+    private @NotNull URI didRename(VirtualFile virtualParentFile, String oldFileName, VirtualFile virtualNewFile) {

```

Comment:
I replaced `didClose` (https://github.com/redhat-developer/lsp4ij/pull/456/commits/f0f008d977f2450f0154217bd810a40959ef7018) with `didRename`, instead of making a separate `didOpen` method.

This is because I didn't think it made sense to have a stand-alone `didOpen` method for the renaming action. A `didOpen` action should only be triggered if we know that a previous document was closed. Otherwise, a rename or move on a non-open document in the IDE might also trigger `didOpen`.

Reply:
Good catch!